app/queries/find_latest_matching_subscription.rb

```diff
@@ -0,0 +1,11 @@
+class FindLatestMatchingSubscription
+  def self.call(original_subscription)
+    subscriber_list_id = original_subscription.subscriber_list_id
+    subscriber_id = original_subscription.subscriber_id
+    frequency = original_subscription.frequency
+    Subscription.
+      where(subscriber_list_id: subscriber_list_id, subscriber_id: subscriber_id, frequency: frequency).
```

Comment:
Do you need to check here whether the subscription is still active?

Reply:
Nope, just want the latest subscription that matches the email address, subscriber list and frequency of the original subscription. Can be active or inactive.

If inactive, we'll display "You're already unsubscribed".